src/type-layout.md

```diff
@@ -162,7 +163,24 @@ representation will not change the layout of `Inner`.
 Nominal types without a `repr` attribute have the default representation.
 Informally, this representation is also called the `rust` representation.
 
-There are no guarantees of data layout made by this representation.
+There are very few data layout guarantees made by this representation. The only
+guarantees are:
+
+ 1. The fields of the struct are properly aligned.
+ 2. The fields do not overlap.
+ 3. The alignment of the struct is not less than the alignment of any of its
+    fields.
```

Comment:
This one reads weird to me. Perhaps something like "The minimum alignment of the struct is at least the maximum alignment of its fields.